tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -207,6 +230,94 @@ void diagnostics_test() {
     }
 }
 
+template <template <class...> class TupleLike>
+void tuple_like_test() {
+    {
+        constexpr TupleLike<int, int> t1{1, 1};
+        constexpr TupleLike<int, int> t1_equal{1, 1};
+        constexpr TupleLike<int, int> t2{2, 1};
+
+        static_assert(spaceship_test<std::strong_ordering>(t1, t1_equal, t2));
+    }
+    {
+        TupleLike<int, int> t1{1, 1};
+        TupleLike<int, int> t1_equal{1, 1};
+        TupleLike<int, int> t2{2, 1};
+
+        spaceship_test<std::strong_ordering>(t1, t1_equal, t2);
+    }
+    {
+        constexpr TupleLike<int, double> t1{1, 1.0};
+        constexpr TupleLike<int, double> t1_equal{1, 1.0};
+        constexpr TupleLike<int, double> t2{2, 1.0};
+
+        static_assert(spaceship_test<std::partial_ordering>(t1, t1_equal, t2));
+    }
+    {
+        TupleLike<int, double> t1{1, 1.0};
+        TupleLike<int, double> t1_equal{1, 1.0};
+        TupleLike<int, double> t2{2, 1.0};
+
+        spaceship_test<std::partial_ordering>(t1, t1_equal, t2);
+    }
+}
+
+template <auto SmallVal, decltype(SmallVal) EqualVal, decltype(EqualVal) LargeVal>
+void optional_test() {
+    using ReturnType = std::compare_three_way_result_t<decltype(SmallVal), decltype(LargeVal)>;
+
+    {

```

Comment:
Ditto "it's less code to maintain if you duplicate at the call site instead of duplicating each test case."